cmake_modules/FindEXAMPLE_PLUGIN_DEP.cmake

```diff
@@ -0,0 +1,92 @@
+  #in which case this can be extracted such that a minimum version check can be conducted at cmake
+  #configuration time. In this example the following is being extracted from a C header file:
+  #"#define EXAMPLE_PLUGIN_DEP_MAJOR 5"
+  #"#define EXAMPLE_PLUGIN_DEP_MINOR 4"
+  #"#define EXAMPLE_PLUGIN_DEP_PATCH 2"
+  #The minimum version to requirement is made visible to this cmake file from
+  #HPCC/plugins/exampleplugin/CMakeLists.txt:ADD_PLUGIN(EXAMPLEPLUGIN PACKAGES EXAMPLE_PLUGIN_DEP MINVERSION 4.6.2)
+  IF  (EXAMPLE_PLUGIN_DEP_INCLUDE_DIR)
+    #MAJOR
+    FILE (STRINGS "${EXAMPLE_PLUGIN_DEP_INCLUDE_DIR}/hpcc-example-plugin-deps.h" major REGEX "#define EXAMPLE_PLUGIN_DEP_MAJOR")
+    STRING (REGEX REPLACE "#define EXAMPLE_PLUGIN_DEP_MAJOR " "" major "${major}")
+    STRING (REGEX REPLACE "\"" "" major "${major}")
+    #MINOR
+    FILE (STRINGS "${EXAMPLE_PLUGIN_DEP_INCLUDE_DIR}/deps.h" minor REGEX "#define EXAMPLE_PLUGIN_DEP_MINOR")
+    STRING (REGEX REPLACE "#define EXAMPLE_PLUGIN_DEP_MINOR " "" minor "${minor}")
+    STRING (REGEX REPLACE "\"" "" minor "${minor}")
+    #PATCH
+    FILE (STRINGS "${EXAMPLE_PLUGIN_DEP_INCLUDE_DIR}/deps.h" patch REGEX "#define EXAMPLE_PLUGIN_DEP_PATCH")
+    STRING (REGEX REPLACE "#define EXAMPLE_PLUGIN_DEP_PATCH " "" patch "${patch}")
+    STRING (REGEX REPLACE "\"" "" patch "${patch}")
+
+    SET (EXAMPLE_PLUGIN_DEP_VERSION_STRING "${major}.${minor}.${patch}")
+    IF ("${EXAMPLE_PLUGIN_DEP_VERSION_STRING}" VERSION_LESS "${EXAMPLE_PLUGIN_DEP_FIND_VERSION}")
+      SET(MSG "libhpcc-example-plugin-deps version '${EXAMPLE_PLUGIN_DEP_VERSION_STRING}' incompatible with min version>=${EXAMPLE_PLUGIN_DEP_FIND_VERSION}")
+    ELSE()
+      SET (EXAMPLE_PLUGIN_DEP_VERSION_OK 1)
+      SET (MSG "${DEFAULT_MSG}")
+    ENDIF()
+  ENDIF()
+
+  #The following three lines are used for building this example as part of the HPCC regression suite.
+  #They are not intended to be present as part of this example itself and should be removed.
+  SET (EXAMPLE_PLUGIN_DEP_LIBRARIES "jlib")
+  SET (EXAMPLE_PLUGIN_DEP_INCLUDE_DIR "./")
+  SET (EXAMPLE_PLUGIN_DEP_VERSION_OK 1)
+
+  #If the following three variables (EXAMPLE_PLUGIN_DEP_LIBRARIES, EXAMPLE_PLUGIN_DEP_INCLUDE_DIR, &
+  #EXAMPLE_PLUGIN_DEP_VERSION_OK) have not been set, MAKE_EXAMPLEPLUGIN will be set false when
+  #returning to HPCC/plugins/exampleplugin/CMakeLists.txt and thus this plugin will not be built
+  #and a message will be given explaining which of three where missing or give the above version
+  #incompatible message.
+  include(FindPackageHandleStandardArgs)
+  find_package_handle_standard_args(example_plugin_dep ${MSG}
+    EXAMPLE_PLUGIN_DEP_LIBRARIES
+    EXAMPLE_PLUGIN_DEP_INCLUDE_DIR
+    EXAMPLE_PLUGIN_DEP_VERSION_OK
+  )
+
+  #I don't believe this is needed, why do/did we do this?
+  MARK_AS_ADVANCED(EXAMPLE_PLUGIN_DEP_INCLUDE_DIR EXAMPLE_PLUGIN_DEP_LIBRARIES)
```

Comment:
Yes.  The find packages is going to create cache variables and we don't want those showing up as configurable in the cache.